drv/sidecar-front-io/sidecar_qsfp_x32_controller_regs.json

```diff
@@ -380,10 +400,41 @@
                 },
                 {
                     "type": "reg",
-                    "inst_name": "PHY_SMI_STATUS",
+                    "inst_name": "PHY_OSC",
                     "addr_offset": 2,
                     "regwidth": 8,
                     "min_accesswidth": 8,
+                    "children": [
+                        {
+                            "type": "field",
+                            "inst_name": "VALID",
+                            "lsb": 0,
+                            "msb": 0,
+                            "reset": 0,
+                            "sw_access": "rw",
+                            "se_onread": null,
+                            "se_onwrite": null,
+                            "desc": "Bit indicating whether or not the GOOD bit is valid"
+                        },
+                        {
+                            "type": "field",
+                            "inst_name": "GOOD",
+                            "lsb": 1,
+                            "msb": 1,
+                            "reset": 0,
+                            "sw_access": "rw",
+                            "se_onread": null,
+                            "se_onwrite": null,
+                            "desc": "Bit indicating whether or not the oscillator is operating nominal"
```

Comment:
Okay, we're actually using this field to (persistently) pass data from one task to the other, which is a little weird but probably okay.

Reply:
Yes, let me elaborate on this a bit. One of my design goals for the sequencer task is that it (or the SP as a whole for that matter) can safely be restarted without interrupting Tofino once the main- and front IO board controllers have been initialized and configured. Hence the external thermal sensor, sequencer logic in the controller etc. There are caveats here that the bitstream versions of those controllers don't change, the clock generator configuration does not change etc., but many times it should be possible to leave Tofino running.

In order to stay true to this goal I needed a more persistent place than sequencer RAM to store the result of the question whether or not the PHY oscillator came out of reset correctly on the last board power up. Otherwise it would either have to take a guess or proactively power cycle the front IO board, interrupting the transceivers. Storing this bit of information with the front IO controller which manages the PHY felt like a reasonable thing to do since it'll automatically get reset if power to the board is cycled, the bitstream of the controller is reloaded or the design is reset (in which case I think the PHY gets reset too).

We're not actually using it to pass data between tasks though. `monorail-server` tells the sequencer that it found the PHY functional, at which point we assume the oscillator is good, and the sequencer then stores it in the front IO board controller (using an API from the `sidecar-front-io-board` crate).